src/main/java/com/example/decafe/Machine.java

```diff
@@ -108,84 +135,102 @@ public void doProgressBarAnimation(Timer productionTimer, ImageView machineImage
                 new TimerTask() {
                     @Override
                     public void run() {
+
                         // After a certain time was reached change the Image of the Machine
                         machineImageView.setImage(imageProductProduced);
+
                         // And make the Machine clickable again
                         machineImageView.setDisable(false);
+
                         // And stop all timers
                         task.stop();
                         timelineBar.stop();
                         productionTimer.cancel();
                     }
                 },
-                this.duration* 1000L
+                this.duration * 1000L
         );
     }
 
     // Method to display a Product / change the state of a Machine
-    public void displayProduct (ImageView waiterImageView, ImageView machineImageView, Player cofiBrew, ProgressBar machineProgressBar) throws FileNotFoundException {
+    public void displayProduct(
+            ImageView waiterImageView,
+            ImageView machineImageView,
+            Player cofiBrew,
+            ProgressBar machineProgressBar
+    ) throws FileNotFoundException {
+
         // create new Timer object
         Timer productionTimer = new Timer();
+
         // Set default image of Waiter and Machine
         String imageMachine = this.filenameImageMachineWithProduct;
         String imageCofi = cofiBrew.getFilenameImageWithoutProduct();
+
         // Set boolean got produced to false - used to check if a product was produced when clicked on Machine
```

Comment:
Do we really need it ? we can know it from this.produced

Reply:
Not the same if this.produced && cofiBrew.getProductInHand().equals(Product.NONE)